repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3483,17 +3487,45 @@ public void deleteClassificationOnlyPropagation(String deletedEdgeId, String cla
             return;
         }
 
-        processClassificationDeleteOnlyPropagation(currentClassificationVertex, edge);
+        processClassificationDeleteOnlyPropagation(currentClassificationVertex, relationshipGuid, isTermEntityEdge(edge));
     }
 
+    public void deleteClassificationOnlyPropagation(String classificationId, String referenceVertexId, boolean isTermEntityEdge) throws AtlasBaseException {
+        AtlasVertex classificationVertex = graph.getVertex(classificationId);
+        AtlasVertex referenceVertex = graph.getVertex(referenceVertexId);
 
-    private void processClassificationDeleteOnlyPropagation(AtlasVertex currentClassificationVertex, AtlasEdge edge) throws AtlasBaseException {
+        if (classificationVertex == null) {
+            LOG.warn("Classification Vertex with ID {} is not present or Deleted", classificationId);
+            return;
+        }
+        /*
+            If reference vertex is deleted, we can consider that as this connected vertex was deleted
+             some other task was created before it to remove propagations. No need to execute this task.
+         */
+        if (referenceVertex == null) {
+            LOG.warn("Reference Vertex is deleted {}", referenceVertexId, classificationId);
```

Comment:
Minor: classificationId will not be printed in logs here

Reply:
Yep, need to update this

Reply:
Updated this, please check it once